clientv3/lease.go

```diff
@@ -98,12 +87,14 @@ type Lease interface {
 	// TimeToLive retrieves the lease information of the given lease ID.
 	TimeToLive(ctx context.Context, id LeaseID, opts ...LeaseOption) (*LeaseTimeToLiveResponse, error)
 
-	// KeepAlive keeps the given lease alive forever.
-	KeepAlive(ctx context.Context, id LeaseID) (<-chan *LeaseKeepAliveResponse, error)
+	// KeepAlive keeps the given lease alive forever. If the keepalive response posted to
+	// the channel is not consumed immediately, the lease client will continue sending keep alive requests
+	// to the etcd server at least every second until latest response is consumed.
+	KeepAlive(ctx context.Context, id LeaseID) LeaseKeepAliveChan
```

Comment:
> Are there docs on this? 

we do not. and, yes, we should. 

>  If the policy is to have it compile but the underlying behavior may change across minor revisions, I don't think that is a meaningful policy whatsoever.

i agree.

> If this is going by semver policy, then this change will violate minor bump behavior no matter what and will need a compat layer instead.

ok. probably the compat layer is the way to go for this release. we need to start thinking about breaking client and server versioning in the next release.